velox/exec/HashProbe.cpp

```diff
@@ -1431,21 +1472,23 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
     static const char* kPassed = "passed";
 
     if (nullAware_) {
-      leftSemiProjectIsNull_.resize(numRows);
-      leftSemiProjectIsNull_.clearAll();
+      leftSemiProjectIsNull_.resize(numRows + offset, false);
+      if (offset == 0) {
```

Comment:
Since left semi does not filter probe rows, we should always get `offset == 0` here, no?  If that's the case just do `VELOX_CHECK_EQ(offset, 0)`; otherwise `leftSemiProjectIsNull_` should be initialized outside the loop.

Reply:
I actually observe the case `offset > 0` in UT.
I moved the intialization for `leftSemiProjectIsNull_` outside the loop. Thanks.

Reply:
Curious when does this happen?

Reply:
The test case I added for semi project join can repro it.
`SELECT t0, t1, t0 IN (SELECT u0 FROM u WHERE (t1 + u1) % 3 = 0) FROM t`
I think we also need to get the results by function `listJoinResults` and `evalFilter` for left semi join with filter, when we merge small vectors, we need to set correct offset for `leftSemiProjectIsNull_`.

Reply:
I see, there is still a filter, thanks